### PR TITLE
Add `/cex` command

### PR DESCRIPTION
**Pull request by tal5**
## Additions

- `/cex` command - executes a single DenizenScript command.
- `/cexq` command - identical to `/cex`, but doesn't output debug to the chat.

## Notes

- The tab-complete logic is mostly the same as the server's, with slight modifications to match the client / utilize some extra tools from brigadier.

### CODE REVIEW

File: src/main/java/com/denizenscript/clientizen/util/CexCommand.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        dispatcher.register(literal("cexq").then(argument("command", greedyString())
+                .suggests(suggestionProvider)
+                .executes(context -> execute(context, true))));
+    }
+
+    public static int execute(CommandContext<FabricClientCommandSource> context, boolean silent) {
+        ExCommandHelper.runString("CEXCOMMAND", getString(context, "command"), null, silent ? null : queue -> {
+            queue.debugOutput = debug -> context.getSource().sendFeedback(Text.literal(debug.replace("<FORCE_ALIGN>", "")));
+        });
+        return Command.SINGLE_SUCCESS;
```
Comment (acikek) [inline src/main/java/com/denizenscript/clientizen/util/CexCommand.java:48]: nitpick, but no one does this, you can just return 1

File: src/main/java/com/denizenscript/clientizen/util/CexCommand.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static com.mojang.brigadier.arguments.StringArgumentType.getString;
+import static com.mojang.brigadier.arguments.StringArgumentType.greedyString;
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.argument;
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.literal;
+
+public class CexCommand {
```
Comment (acikek) [inline src/main/java/com/denizenscript/clientizen/util/CexCommand.java:32]: Might not be relevant due to my other suggestion, but you should name this something like `ClientExecuteCommand`

File: src/main/java/com/denizenscript/clientizen/util/CexCommand.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+    public static void register(CommandDispatcher<FabricClientCommandSource> dispatcher) {
+        final DenizenCommandsSuggestionProvider suggestionProvider = new DenizenCommandsSuggestionProvider();
```
Comment (acikek) [inline src/main/java/com/denizenscript/clientizen/util/CexCommand.java:35]: IS there a reason for this being final when nothing else is?

File: src/main/java/com/denizenscript/clientizen/util/CexCommand.java [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+        dispatcher.register(literal("cex").then(argument("command", greedyString())
+                .suggests(suggestionProvider)
+                .executes(context -> execute(context, false))));
+        dispatcher.register(literal("cexq").then(argument("command", greedyString())
+                .suggests(suggestionProvider)
+                .executes(context -> execute(context, true))));
```
Comment (acikek) [inline src/main/java/com/denizenscript/clientizen/util/CexCommand.java:41]: One of these could be extracted to a method wherein you'd call them here like `register("cex", false);`

File: src/main/java/com/denizenscript/clientizen/util/CexCommand.java [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+    }
+
+    public static int execute(CommandContext<FabricClientCommandSource> context, boolean silent) {
+        ExCommandHelper.runString("CEXCOMMAND", getString(context, "command"), null, silent ? null : queue -> {
+            queue.debugOutput = debug -> context.getSource().sendFeedback(Text.literal(debug.replace("<FORCE_ALIGN>", "")));
```
Comment (acikek) [inline src/main/java/com/denizenscript/clientizen/util/CexCommand.java:46]: I've been seeing the removal of special codes like `<FORCE_ALIGN>` in a few places, maybe we could extract these to the utility methods?

File: src/main/java/com/denizenscript/clientizen/util/CexCommand.java [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+        });
+        return Command.SINGLE_SUCCESS;
+    }
+
+    public static class DenizenCommandsSuggestionProvider implements SuggestionProvider<FabricClientCommandSource> {
```
[the remark this answers is not in the document]
Comment (tal5) [reply src/main/java/com/denizenscript/clientizen/util/CexCommand.java]: Good idea, done

File: src/main/java/com/denizenscript/clientizen/util/CexCommand.java (hunk 1 above)
Comment (tal5) [reply src/main/java/com/denizenscript/clientizen/util/CexCommand.java]: Fair enough, changed it

File: src/main/java/com/denizenscript/clientizen/util/CexCommand.java (hunk 2 above)
Comment (tal5) [reply src/main/java/com/denizenscript/clientizen/util/CexCommand.java]: No longer relevant due to other suggestion

File: src/main/java/com/denizenscript/clientizen/util/CexCommand.java (hunk 3 above)
Comment (tal5) [reply src/main/java/com/denizenscript/clientizen/util/CexCommand.java]: No longer relevant due to other suggestion

File: src/main/java/com/denizenscript/clientizen/util/CexCommand.java (hunk 5 above)
Comment (tal5) [reply src/main/java/com/denizenscript/clientizen/util/CexCommand.java]: I don't think it's worth a util method tbh, it's only `<FORCE_ALIGN>` and it's only done in places that handle debug output, which it's unlikely we'll have too many of

**Comment by acikek** [comment]
LGTM

**Pull request merged**